teacher_spec.rb

```diff
@@ -0,0 +1,19 @@
+require './teacher'
+
+describe Teacher do
+  context 'when testing teacher class' do
+    before :all do
+      @teacher = Teacher.new('Biology', 55, 'Taye G.')
+    end
+
+    it 'should have age, name, specialization, parent_permission, rentals' do
+      expect(@teacher).to have_attributes(age: 55)
+      expect(@teacher).to have_attributes(name: 'Taye G.')
+      expect(@teacher).to have_attributes(specialization: 'Biology')
+      expect(@teacher.rentals).to be_empty
+    end
+    it 'should return true when can_use_service? is called' do
+      expect(@teacher.can_use_services?).to be true
+    end
+  end
+end
```

Comment:
- [OPTIONAL] Although it is not in your requirements, it would be nice to put this spec file in the `spec` directory. Putting all test files in one folder simplifies test organization, making it easier to locate and manage tests while facilitating consistent test configuration and easy test execution.